sdk/rust/nym-sdk/examples/socks5.rs

```diff
@@ -7,27 +7,17 @@ async fn main() {
     println!("Connecting receiver");
     let mut receiving_client = mixnet::MixnetClient::connect_new().await.unwrap();
 
-    let config = mixnet::Config {
-        socks5_service_provider: Some(receiving_client.nym_address().to_string()),
-        ..Default::default()
-    };
+    let socks5_config = mixnet::Socks5::new(receiving_client.nym_address().to_string());
     let sending_client = mixnet::MixnetClientBuilder::new()
-        .config(config)
+        .socks5_config(socks5_config)
         .build::<mixnet::EmptyReplyStorage>()
         .await
         .unwrap();
 
     println!("Connecting sender");
-    let mut sending_client = sending_client.connect_to_mixnet().await.unwrap();
-    // wait until socks5 server is started
-    println!("Wait 5 seconds for the socks5 setup to be done");
-    std::thread::sleep(std::time::Duration::from_secs(5));
+    let mut sending_client = sending_client.connect_to_mixnet_via_socks5().await.unwrap();
```

Comment:
The name `connect_to_mixnet_via_socks5` kinda makes the user think that the client is connecting to the mixnet via socks5 when it's connecting to the mixnet and exposing a socks5 endpoint. But I can't come up with a better name .. :man_shrugging:

Reply:
While that is true, and there is technically a way for the user to talk directly to the mixnet, I specifically removed the communication channels with the mixnet from the user interface for this type of connection, so that the socks5 port is the only way for it to use the mixnet. But I agree, this is a bit of a head spinner and will need to be well documented. The `socks5` example is one way to start that, but more docs around that will probably also be needed.